### ISSUE TITLE
Unbound: disabling the first DNS override entry results in invalid configuration file

### ISSUE THREAD

**Issue by marinbernard-pep06** [opnsense/core#5798]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [ x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [ x] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

We use Unbound domain overrides to redirect requests matching our DNS domains to our own servers. Each domain name has several override entries. When the first of entry is disabled in the OPNsense Web UI, the resulting config file (`/var/unbound/etc/domainoverrides.conf`) becomes invalid: zone headers and names are not included.

For instance, we have three overrides for domain `intranet.pep06.fr`. When all override entries are enabled, the config file includes the following forward zone:

```
forward-zone:
  name: "intranet.pep06.fr"
  forward-addr: 10.6.136.254
  forward-addr: 10.6.184.254
  forward-addr: 10.6.104.254
```

Disabling entry `10.6.136.254` in the Web UI makes the zone header disappear:

```
  forward-addr: 10.6.184.254
  forward-addr: 10.6.104.254
```

**To Reproduce**

1. Create several DNS override entries related to a single domain
2. Disable the first override entry
3. Save configuration
4. The resulting config file is invalid

**Expected behavior**

The forward zone header and name should appear as long as a single override entry is enabled.

**Describe alternatives you considered**

None

**Screenshots**

None

**Relevant log files**

None

**Additional context**

AFAIK, the bug was already present in 22.1.7.

**Environment**

Irrelevant

**Comment by AdSchellevis** [comment]
@marinbernard-pep06 can you try https://github.com/opnsense/core/commit/cdcf22e80dc0ff5a6eff1b019941d7ff48cede74 ?

**Comment by marinbernard-pep06** [comment]
@AdSchellevis  I confirm the patch fixes the issue. Many thanks!

**Comment by tinus-github** [comment]
I just noticed this issue after upgrading to 23.7. It looks like this got reverted in commit a556df4

**Comment by fichtner** [comment]
What is a556df4? Doesn’t exist.

**Comment by tinus-github** [comment]
It is this commit:

https://github.com/opnsense/core/commit/a556df4a85df41db80b4b4ec22e9f9bfc79f2925

If you look at the current version of the file at

https://github.com/opnsense/core/blob/master/src/opnsense/service/templates/OPNsense/Unbound/core/domainoverrides.conf

you can see that the change made to fix this issue ( #5798 ) which is https://github.com/opnsense/core/commit/cdcf22e80dc0ff5a6eff1b019941d7ff48cede74 has accidentally been reverted by a later update.

**Comment by AdSchellevis** [comment]
@tinus-github looks like a regression indeed, should be fixed in https://github.com/opnsense/core/commit/8de0986673bf80131a5af269c0e969e0fb835c0a
